modules/util/portal-tools-sample-sql-builder/src/main/java/com/liferay/portal/tools/sample/sql/builder/DataFactory.java

```diff
@@ -602,14 +625,129 @@ public long getDefaultDLDDMStructureId() {
 		return _defaultDLDDMStructureId;
 	}
 
+	public long getDefaultObjectFolderId() {
+		return _defaultObjectFolderId;
+	}
+
 	public long getDLFileEntryClassNameId() {
 		return getClassNameId(DLFileEntry.class);
 	}
 
+	public String getDynamicObjectDefinitionTableCreateSQL(
+		ObjectDefinitionModel objectDefinitionModel,
+		List<ObjectFieldModel> objectFieldModels) {
+
+		List<ObjectField> objectFields = new ArrayList<>(
+			objectFieldModels.size());
+
+		for (ObjectFieldModel objectFieldModel : objectFieldModels) {
+			if (objectFieldModel.isSystem()) {
+				continue;
+			}
+
+			objectFields.add((ObjectField)objectFieldModel);
+		}
+
+		DynamicObjectDefinitionTable dynamicObjectDefinitionTable =
+			new DynamicObjectDefinitionTable(
+				(ObjectDefinition)objectDefinitionModel, objectFields,
+				objectDefinitionModel.getDBTableName());
+
+		return dynamicObjectDefinitionTable.getCreateTableSQL();
+	}
+
+	public String getExtensionDynamicObjectDefinitionTableCreateSQL(
+		ObjectDefinitionModel objectDefinitionModel) {
+
+		String tableName = objectDefinitionModel.getDBTableName();
+
+		if (!tableName.endsWith("_")) {
```

Comment:
Is there a way to reuse our existing logic here? Using methods like `objectDefinitionImpl#getExtensionDBTableName`.
I'm afraid that it will break the tests if we change this logic in the object code, as the change will not be reflected here.

Reply:
Yes we can use this!